lib/stdlib/doc/src/lists.xml

```diff
@@ -323,6 +323,28 @@ flatmap(Fun, List1) ->
       </desc>
     </func>
 
+    <func>
+      <name name="foldlwhile" arity="3" since=""/>
+      <fsummary>Fold a function over a list until <anno>halt</anno> is returned</fsummary>
+      <desc>
+        <p>Calls <c><anno>Fun</anno>(<anno>Elem</anno>, <anno>AccIn</anno>)</c>
+          on successive elements <c>A</c> of <c><anno>List</anno></c>, starting
+          with <c><anno>AccIn</anno> == <anno>Acc0</anno></c>.
+          <c><anno>Fun</anno>/2</c> must return a tuple <c>{<anno>cont</anno>, <anno>Acc</anno>}</c>
+          to continue or <c>{<anno>halt</anno>, <anno>Acc</anno>}</c>
+          to stop the folding. The second element of the tuple is the accumulator, which is
+          passed to the next call. The function returns the final value of
+          the accumulator. <c><anno>Acc0</anno></c> is returned if the list is
+          empty.</p>
+        <p><em>Example:</em></p>
+        <pre>
+> <input>lists:foldlwhile(fun(X, Sum) when X&lt;3 -> {cont, X + Sum}; (X, Sum) -> {halt, X + Sum} end, 0, [1,2,3,4,5]).</input>
+6
```

Comment:
I don't think this example is obvious. I would expect the iteration to return the sum of elements with a value less than 3. In this case 1+2. Since it halt on 3 I don't think it is obvious to add 3 to the result in the `{halt, X + Sum}`.
It would have been more intuitive if the fun would have been `fun(X,Sum) when X=<3 -> {cont, X + Sum}; (X, Sum) -> {halt, X + Sum} end`. In this case it feels natural that the value 3 is included in the sum.

Reply:
thanks for the suggestion, just fixed